charts/spire/charts/spire-server/templates/post-install-hook.yaml

```diff
@@ -0,0 +1,85 @@
+{{- if eq (.Values.controllerManager.enabled | toString) "true" }}
+{{- if eq .Values.controllerManager.validatingWebhookConfiguration.failurePolicy "Fail" }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "spire-server.serviceAccountName" . }}-post-install
+  namespace: {{ include "spire-server.namespace" . }}
+  labels:
+    {{- include "spire-server.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-succeeded, hook-failed
```

Comment:
hmm.. what happens on hook-failure? if it deletes the job, no one might notice it has failed and the webhook is still Ignore? or does the helm upgrade fail and its ok cause the user will notice?